Content.Server/Atmos/EntitySystems/GasTankSystem.cs

```diff
@@ -12,227 +12,235 @@
-        [Dependency] private readonly UserInterfaceSystem _ui = default!;
-        [Dependency] private readonly IRobustRandom _random = default!;
-        [Dependency] private readonly ThrowingSystem _throwing = default!;
-        [Dependency] private readonly IConfigurationManager _cfg = default!;
-
-        private const float TimerDelay = 0.5f;
-        private float _timer = 0f;
-        private const float MinimumSoundValvePressure = 10.0f;
-        private float _maxExplosionRange;
-
-        public override void Initialize()
-        {
-            base.Initialize();
-            SubscribeLocalEvent<GasTankComponent, EntParentChangedMessage>(OnParentChange);
-            SubscribeLocalEvent<GasTankComponent, GasAnalyzerScanEvent>(OnAnalyzed);
-            SubscribeLocalEvent<GasTankComponent, PriceCalculationEvent>(OnGasTankPrice);
-            Subs.CVar(_cfg, CCVars.AtmosTankFragment, UpdateMaxRange, true);
-        }
+        base.Initialize();
+        SubscribeLocalEvent<GasTankComponent, EntParentChangedMessage>(OnParentChange);
+        SubscribeLocalEvent<GasTankComponent, GasAnalyzerScanEvent>(OnAnalyzed);
+        SubscribeLocalEvent<GasTankComponent, PriceCalculationEvent>(OnGasTankPrice);
+        SubscribeLocalEvent<GasTankComponent, BoundUIOpenedEvent>(OnUIOpened);
+        Subs.CVar(_cfg, CCVars.AtmosTankFragment, UpdateMaxRange, true);
+    }
 
-        private void UpdateMaxRange(float value)
-        {
-            _maxExplosionRange = value;
-        }
+    private void OnUIOpened(Entity<GasTankComponent> ent, ref BoundUIOpenedEvent args)
+    {
+        UpdateUserInterface(ent);
+    }
 
-        public override void UpdateUserInterface(Entity<GasTankComponent> ent)
-        {
-            var (owner, component) = ent;
-            _ui.SetUiState(owner, SharedGasTankUiKey.Key,
-                new GasTankBoundUserInterfaceState
-                {
-                    TankPressure = component.Air?.Pressure ?? 0,
-                });
-        }
+    private void UpdateMaxRange(float value)
+    {
+        _maxExplosionRange = value;
+    }
 
-        private void OnParentChange(EntityUid uid, GasTankComponent component, ref EntParentChangedMessage args)
-        {
-            // When an item is moved from hands -> pockets, the container removal briefly dumps the item on the floor.
-            // So this is a shitty fix, where the parent check is just delayed. But this really needs to get fixed
-            // properly at some point.
-            component.CheckUser = true;
-        }
+    public override void UpdateUserInterface(Entity<GasTankComponent> ent)
+    {
+        var pressure = ent.Comp.OutputPressure;
+        var internalsConnected = ent.Comp.IsConnected;
+
+        UI.ServerSendUiMessage(ent.Owner,
+            SharedGasTankUiKey.Key,
+            new GasTankUpdateMessage(pressure,
+                ent.Comp.Air.Pressure,
+                ent.Comp.IsValveOpen,
+                internalsConnected));
```

Comment:
this is not how we do UI states, prefer to read from the shared components directly on the client side (i.e. hook into the AfterAutoHandleStateEvent that's already being handled by GasTankSystem on the client & read data from the components as necessary), or if that's not possible, use the usual BUI state infrastructure that you're replacing